x-pack/plugins/infra/public/alerting/metric_threshold/components/alert_details_app_section.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { Rule } from '@kbn/alerting-plugin/common';
+import { MetricThresholdRuleTypeParams } from '..';
+import { generateUniqueKey } from '../lib/generate_unique_key';
+import { MetricsExplorerChartType } from '../../../pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options';
+import { ExpressionChart } from './expression_chart';
+import { useSourceViaHttp } from '../../../containers/metrics_source/use_source_via_http';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+
+// TODO Use a generic props for app sections https://github.com/elastic/kibana/issues/152690
+interface AppSectionProps {
+  rule: Rule<
+    MetricThresholdRuleTypeParams & {
+      filterQueryText?: string;
+      groupBy?: string | string[];
+    }
+  >;
+}
+
+export function AlertDetailsAppSection({ rule }: AppSectionProps) {
+  const { http, notifications } = useKibanaContextForPlugin().services;
+  const { source, createDerivedIndexPattern } = useSourceViaHttp({
```

Comment:
**question**: could you explain why you used the `useSourceViaHttp` hook instead of the `useSourceContext` one please? :) 

We have a [wip PR](https://github.com/elastic/kibana/pull/152840) that, since they are logically identical, removes the `useSourceViaHttp` hook and replaces its usage with the `useSourceContext` hook, so that we keep a single source of truth. The passed options to `useSourceViaHttp` in this use case are the same used inside the `useSourceContext` implementation.

Reply:
I wasn't aware of `useSourceContext`, I will use `useSourceContext` then :)

Reply:
We had a discussion offline, and we decided to change this in Marco's PR, so I'll leave it as it is for now.